cdap-security/src/main/java/co/cask/cdap/security/authorization/RemoteAuthorizationEnforcer.java

```diff
@@ -119,15 +183,82 @@ private boolean doEnforce(AuthorizationPrivilege authorizationPrivilege) throws
     return HttpURLConnection.HTTP_OK == remoteClient.execute(request).getResponseCode();
   }
 
-  private Set<? extends EntityId> checkVisibility(VisibilityRequest visibilityRequest) throws IOException {
+  private Set<? extends EntityId> visibilityCheckCall(VisibilityRequest visibilityRequest) throws IOException {
     HttpRequest request = remoteClient.requestBuilder(HttpMethod.POST, "isVisible")
       .withBody(GSON.toJson(visibilityRequest))
       .build();
     return GSON.fromJson(remoteClient.execute(request).getResponseBodyAsString(), SET_ENTITY_TYPE);
   }
 
-  @VisibleForTesting
-  public Map<AuthorizationPrivilege, Boolean> cacheAsMap() {
-    return Collections.unmodifiableMap(authPolicyCache.asMap());
+  private Map<VisibilityKey, Boolean> loadVisibility(Iterable<? extends VisibilityKey> keys) throws IOException {
+    if (!keys.iterator().hasNext()) {
+      return Collections.emptyMap();
+    }
+
+    Principal principal = keys.iterator().next().getPrincipal();
```

Comment:
Shall we put a comment saying by the current implementation, the principal will not be a set?